Support Ghidra 12.1

Build errors:

```
> Task :compileJava
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/loader/SwitchLoader.java:102: error: method does not override or implement a method from a supertype
    @Override
    ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/loader/SwitchLoader.java:110: error: no suitable method found for createProgram(ByteProvider,String,Address,String,Language,CompilerSpec,Object)
        Program prog = createProgram(provider, programName, baseAddr, getName(), importerLanguage, importerCompilerSpec, consumer);
                       ^
    method AbstractProgramLoader.createProgram(Address,ImporterSettings) is not applicable
      (actual and formal argument lists differ in length)
    method AbstractProgramLoader.createProgram(ImporterSettings) is not applicable
      (actual and formal argument lists differ in length)
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/loader/SwitchLoader.java:117: error: method loadInto in class AbstractProgramLoader cannot be applied to given types;
            this.loadInto(provider, loadSpec, options, log, prog, monitor);
                ^
  required: Program,ImporterSettings
  found:    ByteProvider,LoadSpec,List<Option>,MessageLog,Program,TaskMonitor
  reason: actual and formal argument lists differ in length
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/loader/SwitchLoader.java:119: error: cannot infer type arguments for Loaded<>
            results = List.of(new Loaded<>(prog, programName, programFolderPath));
                              ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/loader/SwitchLoader.java:132: error: method does not override or implement a method from a supertype
    @Override
    ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java:540: warning: [removal] PLATE_COMMENT in CodeUnit has been deprecated and marked for removal
                        program.getListing().setComment(entry.addr, CodeUnit.PLATE_COMMENT, ipcComment);
                                                                            ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java:540: warning: [removal] setComment(Address,int,String) in Listing has been deprecated and marked for removal
                        program.getListing().setComment(entry.addr, CodeUnit.PLATE_COMMENT, ipcComment);
                                            ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java:553: warning: [removal] REPEATABLE_COMMENT in CodeUnit has been deprecated and marked for removal
                        program.getListing().setComment(entry.addr, CodeUnit.REPEATABLE_COMMENT, entry.fullName);
                                                                            ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java:553: warning: [removal] setComment(Address,int,String) in Listing has been deprecated and marked for removal
                        program.getListing().setComment(entry.addr, CodeUnit.REPEATABLE_COMMENT, entry.fullName);
                                            ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java:585: warning: [removal] REPEATABLE_COMMENT in CodeUnit has been deprecated and marked for removal
                        program.getListing().setComment(vtAddr, CodeUnit.REPEATABLE_COMMENT, name);
                                                                        ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java:585: warning: [removal] setComment(Address,int,String) in Listing has been deprecated and marked for removal
                        program.getListing().setComment(vtAddr, CodeUnit.REPEATABLE_COMMENT, name);
                                            ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java:622: warning: [removal] PLATE_COMMENT in CodeUnit has been deprecated and marked for removal
                    program.getListing().setComment(ipcCmdImplAddr, CodeUnit.PLATE_COMMENT, implComment);
                                                                            ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java:622: warning: [removal] setComment(Address,int,String) in Listing has been deprecated and marked for removal
                    program.getListing().setComment(ipcCmdImplAddr, CodeUnit.PLATE_COMMENT, implComment);
                                        ^
/tmp/tmp.9QxOGVAu4U/Ghidra-Switch-Loader/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java:726: warning: [removal] demangle(String) in DemanglerUtil has been deprecated and marked for removal
        DemangledObject demangledObj = DemanglerUtil.demangle(mangled);
                                                    ^
5 errors
9 warnings

```

--- a/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java
+++ b/src/main/java/adubbz/nx/analyzer/IPCAnalyzer.java
@@ -28,6 +28,7 @@ import ghidra.program.model.address.AddressSetView;
 import ghidra.program.model.address.AddressSpace;
 import ghidra.program.model.data.PointerDataType;
 import ghidra.program.model.listing.CodeUnit;
+import ghidra.program.model.listing.CommentType;
 import ghidra.program.model.listing.Data;
 import ghidra.program.model.listing.Program;
 import ghidra.program.model.mem.Memory;
@@ -537,7 +538,7 @@ public class IPCAnalyzer extends AbstractAnalyzer
                     if (sTableAddr != null)
                     {
                         ipcComment = String.format(ipcComment, sTableAddr.getOffset());
-                        program.getListing().setComment(entry.addr, CodeUnit.PLATE_COMMENT, ipcComment);
+                        program.getListing().setComment(entry.addr, CommentType.PLATE, ipcComment);
                     }
                     
                     ipcTraces = Lists.newArrayList(processFuncTraces.get(processFuncAddr).iterator());
@@ -550,7 +551,7 @@ public class IPCAnalyzer extends AbstractAnalyzer
                 {
                     // For shortened names, leave a comment so the user knows what the original name is
                     if (!entry.fullName.equals(entry.abvName))
-                        program.getListing().setComment(entry.addr, CodeUnit.REPEATABLE_COMMENT, entry.fullName);
+                        program.getListing().setComment(entry.addr, CommentType.REPEATABLE, entry.fullName);
                     
                     Msg.info(this, String.format("Creating label for %s @ 0x%X", entry.abvName, entry.addr.getOffset()));
                     program.getSymbolTable().createLabel(entry.addr, entry.abvName, null, SourceType.IMPORTED);
@@ -582,7 +583,7 @@ public class IPCAnalyzer extends AbstractAnalyzer
                     }
                     else
                     {
-                        program.getListing().setComment(vtAddr, CodeUnit.REPEATABLE_COMMENT, name);
+                        program.getListing().setComment(vtAddr, CommentType.REPEATABLE, name);
                     }
                 }
                 
@@ -619,7 +620,7 @@ public class IPCAnalyzer extends AbstractAnalyzer
                             """;
                     
                     implComment = String.format(implComment, trace.bytesIn, trace.bytesOut, trace.bufferCount, trace.inInterfaces, trace.outInterfaces, trace.inHandles, trace.outHandles);
-                    program.getListing().setComment(ipcCmdImplAddr, CodeUnit.PLATE_COMMENT, implComment);
+                    program.getListing().setComment(ipcCmdImplAddr, CommentType.PLATE, implComment);
                 }
             }
             

--- a/src/main/java/adubbz/nx/loader/SwitchLoader.java
+++ b/src/main/java/adubbz/nx/loader/SwitchLoader.java
@@ -24,9 +24,7 @@ import ghidra.app.util.bin.BinaryReader;
 import ghidra.app.util.bin.ByteProvider;
 import ghidra.app.util.bin.ByteProviderWrapper;
 import ghidra.app.util.importer.MessageLog;
-import ghidra.app.util.opinion.BinaryLoader;
-import ghidra.app.util.opinion.LoadSpec;
-import ghidra.app.util.opinion.LoaderTier;
+import ghidra.app.util.opinion.*;
 import ghidra.framework.model.Project;
 import ghidra.framework.store.LockException;
 import ghidra.program.model.address.Address;
@@ -38,7 +36,6 @@ import ghidra.program.model.lang.Language;
 import ghidra.program.model.lang.LanguageCompilerSpecPair;
 import ghidra.program.model.lang.LanguageID;
 import ghidra.program.model.listing.Program;
-import ghidra.app.util.opinion.Loaded;
 import ghidra.util.Msg;
 import ghidra.util.exception.CancelledException;
 import ghidra.util.task.TaskMonitor;
@@ -100,29 +97,29 @@ public class SwitchLoader extends BinaryLoader
     }
 
     @Override
-    protected List<Loaded<Program>> loadProgram(ByteProvider provider, String programName, Project project, String programFolderPath, LoadSpec loadSpec, List<Option> options, MessageLog log, Object consumer, TaskMonitor monitor) throws IOException, CancelledException
-    {
+    protected List<Loaded<Program>> loadProgram(ImporterSettings settings) throws IOException, CancelledException {
+        LoadSpec loadSpec = settings.loadSpec();
         LanguageCompilerSpecPair pair = loadSpec.getLanguageCompilerSpec();
         Language importerLanguage = getLanguageService().getLanguage(pair.languageID);
         CompilerSpec importerCompilerSpec = importerLanguage.getCompilerSpecByID(pair.compilerSpecID);
 
         Address baseAddr = importerLanguage.getAddressFactory().getDefaultAddressSpace().getAddress(0);
-        Program prog = createProgram(provider, programName, baseAddr, getName(), importerLanguage, importerCompilerSpec, consumer);
+        Program prog = createProgram(settings);
         boolean success = false;
 
         List<Loaded<Program>> results;
 
         try 
         {
-            this.loadInto(provider, loadSpec, options, log, prog, monitor);
+            this.loadInto(prog, settings);
             success = true;
-            results = List.of(new Loaded<>(prog, programName, programFolderPath));
+            results = List.of(new Loaded<Program>(prog, settings));
         }
         finally 
         {
             if (!success) 
             {
-                prog.release(consumer);
+                prog.release(settings);
             }
         }
 
@@ -130,11 +127,9 @@ public class SwitchLoader extends BinaryLoader
     }
 
     @Override
-    protected void loadProgramInto(ByteProvider provider, LoadSpec loadSpec, List<Option> options,
-            MessageLog messageLog, Program program, TaskMonitor monitor)
-                    throws IOException, CancelledException
-    {
+    protected void loadProgramInto(Program program, ImporterSettings settings) throws IOException, CancelledException {
         var space = program.getAddressFactory().getDefaultAddressSpace();
+        var provider = settings.provider();
         
         if (this.binaryType == BinaryType.SX_KIP1)
         {
@@ -161,12 +156,12 @@ public class SwitchLoader extends BinaryLoader
         }
 
         var loader = new NXProgramBuilder(program, provider, adapter);
-        loader.load(monitor);
+        loader.load(settings.monitor());
         
         if (this.binaryType == BinaryType.KIP1)
         {
             // KIP1s always start with a branch instruction at the start of their text
-            loader.createEntryFunction("entry", program.getImageBase().getOffset(), monitor);
+            loader.createEntryFunction("entry", program.getImageBase().getOffset(), settings.monitor());
         }
     }
 

